### PR TITLE
In-memory engine: set boundries when gc

### LINKED ISSUE #17143: In-memory Engine: boundaries are not specified when GC
## Bug Report
https://github.com/tikv/tikv/blob/master/components/region_cache_memory_engine/src/background.rs#L341-L354

boundaries for the range should be specified when gc.

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Close #17143 Ref #16141

What's Changed:

```commit-message
set boundries when gc
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/region_cache_memory_engine/src/background.rs [hunk 1]
```diff
@@ -340,8 +343,9 @@ impl BackgroundRunnerCore {
 
         let mut iter = write_cf_handle.iterator();
         let guard = &epoch::pin();
-        iter.seek_to_first(guard);
-        while iter.valid() {
+        let (start_key, end_key) = encode_key_for_eviction(range);
```
Comment (zhangjinpeng87) [inline components/region_cache_memory_engine/src/background.rs:346]: `eoncode_key_for_eviction` is not a proper name here, here you just want to reuse the implementation of this function, but the semantic is not correct.

Comment (zhangjinpeng87) [reply components/region_cache_memory_engine/src/background.rs:346]: Better to rename `encode_key_for_eviction` to `encode_range_keys` or similar name. The name `encode_key_for_eviction` is not well abstracted, and it is too specific.

**Review by zhangjinpeng87 — APPROVED** [review]
rest LGTM

File: components/region_cache_memory_engine/src/background.rs (hunk 1 above)
Comment (SpadeA-Tang) [reply components/region_cache_memory_engine/src/background.rs:346]: In my another PR, I have change it to `encode_key_for_boundary_with_mvcc`. Also, there's another method called `encode_key_for_boundary_without_mvcc` in that PR.

**Review by overvenus — APPROVED** [review]

**Pull request merged**